eclair-core/src/test/scala/fr/acinq/eclair/wire/protocol/OfferTypesSpec.scala

```diff
@@ -50,72 +42,30 @@ class OfferTypesSpec extends AnyFunSuite {
   }
 
-  test("basic signed offer") {
-    val signedOffer = Offer(TlvStream[OfferTlv](
-      Description("basic signed offer"),
-      NodeId(nodeId))).sign(nodeKey)
-    val encoded = "lno1pgfxyctnd93jqumfvahx2epqdanxvetjrcssxr0juddeytv7nwawhk9nq9us0arnk8j8wnsq8r2e86vzgtfneupe7pqr8k27dajwvrehuprj08ggamld6pgnj9ydp3whx6kz5hjaavh8rhfjzkhyxsjwakelepqxmx26aqhlaslxn8ljn4mtm2cx76xz72kxkc"
-    assert(Offer.decode(encoded).get == signedOffer)
-    assert(signedOffer.checkSignature())
-    assert(signedOffer.amount.isEmpty)
-    assert(signedOffer.description == "basic signed offer")
-    assert(signedOffer.nodeId == nodeId)
-  }
-
   test("offer with amount and quantity") {
     val offer = Offer(TlvStream[OfferTlv](
-      Chains(Seq(Block.TestnetGenesisBlock.hash)),
-      Amount(50 msat),
-      Description("offer with quantity"),
-      Issuer("alice@bigshop.com"),
-      QuantityMin(1),
-      NodeId(nodeKey.publicKey)))
-    val encoded = "lno1qgsyxjtl6luzd9t3pr62xr7eemp6awnejusgf6gw45q75vcfqqqqqqqgqyeq5ym0venx2u3qwa5hg6pqw96kzmn5d968j9q3v9kxjcm9gp3xjemndphhqtnrdak3vqgprcssxr0juddeytv7nwawhk9nq9us0arnk8j8wnsq8r2e86vzgtfneupe"
+      OfferChains(Seq(Block.TestnetGenesisBlock.hash)),
+      OfferAmount(50 msat),
+      OfferDescription("offer with quantity"),
+      OfferIssuer("alice@bigshop.com"),
+      OfferQuantityMax(0),
+      OfferNodeId(nodeKey.publicKey)))
+    val encoded = "lno1qgsyxjtl6luzd9t3pr62xr7eemp6awnejusgf6gw45q75vcfqqqqqqqgqyeq5ym0venx2u3qwa5hg6pqw96kzmn5d968jys3v9kxjcm9gp3xjemndphhqtnrdak3gqqkyypsmuhrtwfzm85mht4a3vcp0yrlgua3u3m5uqpc6kf7nqjz6v70qwg"
     assert(Offer.decode(encoded).get == offer)
     assert(offer.amount.contains(50 msat))
-    assert(offer.signature.isEmpty)
     assert(offer.description == "offer with quantity")
     assert(offer.nodeId == nodeId)
     assert(offer.issuer.contains("alice@bigshop.com"))
-    assert(offer.quantityMin.contains(1))
-  }
-
-  test("signed offer with amount and quantity") {
-    val signedOffer = Offer(TlvStream[OfferTlv](
-      Chains(Seq(Block.TestnetGenesisBlock.hash)),
-      Amount(50 msat),
-      Description("offer with quantity"),
-      Issuer("alice@bigshop.com"),
-      QuantityMin(1),
-      NodeId(nodeKey.publicKey))).sign(nodeKey)
-    val encoded = "lno1qgsyxjtl6luzd9t3pr62xr7eemp6awnejusgf6gw45q75vcfqqqqqqqgqyeq5ym0venx2u3qwa5hg6pqw96kzmn5d968j9q3v9kxjcm9gp3xjemndphhqtnrdak3vqgprcssxr0juddeytv7nwawhk9nq9us0arnk8j8wnsq8r2e86vzgtfneupe7pqte5yn8m6mtk5racz9c0hgw6smxhp5t0ns77huttmy4632f6t2ns3jdwkxh9qy9f2eun2329gswcz38dn5f2us4us2r76zxvhkj9uecv"
-    assert(Offer.decode(encoded).get == signedOffer)
-    assert(signedOffer.checkSignature())
-    assert(signedOffer.amount.contains(50 msat))
-    assert(signedOffer.description == "offer with quantity")
-    assert(signedOffer.nodeId == nodeId)
-    assert(signedOffer.issuer.contains("alice@bigshop.com"))
-    assert(signedOffer.quantityMin.contains(1))
-  }
-
-  test("decode invalid offer") {
```

Comment:
We're now missing tests that decode invalid offers and invoice requests.

Reply:
I've removed them because I didn't think they bring much value. For instance in this PR, I've changed a lot of the code and all of these invalidity tests still pass without needing to be updated.
I've added some new tests to replace them: I create a minimal offer / invoice request and check that removing any TLV makes it invalid.